packages/affiliates/libs/uniswap/utils.ts

```diff
@@ -32,3 +24,57 @@ export function convertValuesToString<T>(obj: { [k: string]: any }): T {
     })
   ) as T;
 }
+
+// This looks at a positions liquidity, and calculates the amount across a single tick based on full tick span
+export function liquidityPerTick(params: { liquidity: string; tickLower: string; tickUpper: string }) {
+  const { liquidity, tickLower, tickUpper } = params;
+  assert(BigInt(tickUpper) > BigInt(tickLower), "Upper tick must be > lower tick");
+  return (BigInt(liquidity) / (BigInt(tickUpper) - BigInt(tickLower))).toString();
+}
+
+// this is stronger than position, but compatible, since these fields have been validated to exist
+type ActivePosition = Position & {
+  liquidity: BigNumberish;
+  tickUpper: BigNumberish;
+  tickLower: BigNumberish;
+};
+
+// figure out if a position is active by looking a liquidity and combination of current tick and position bounds
+// this assume the upper tick is not inclusive, meaning if the upper tick matches the current tick, its not active
+export const IsPositionActive = (tick: BigNumberish) => (
+  position: Pick<Position, "liquidity" | "tickUpper" | "tickLower">
+): position is ActivePosition => {
+  assert(position.liquidity, "requires position liquidity");
+  assert(position.tickUpper, "requires position tickUpper");
+  assert(position.tickLower, "requires position tickLower");
+  if (BigInt(position.liquidity.toString()) === 0n) return false;
+  if (BigInt(tick.toString()) >= BigInt(position.tickUpper.toString())) return false;
+  if (BigInt(tick.toString()) < BigInt(position.tickLower.toString())) return false;
+  if (BigInt(position.tickUpper) <= BigInt(position.tickLower)) return false;
+  return true;
+};
+
+// percent: numerator, denominator and scale factor
+export function percent(val: string | BigInt, sum: string | BigInt, scale: string | BigInt = 10n ** 18n) {
+  return (BigInt(val) * BigInt(scale)) / BigInt(sum);
+}
+
+export function percentShares(
+  contributions: { [key: string]: string } = {},
+  sum?: string | BigInt,
+  scale: string | BigInt = 10n ** 18n
```

Comment:
nit: why not make these either strings or BigInts. For all of these methods, can we just require the caller to coerce to BigInt rather than pass around these `|` types?

Reply:
yea its a good point. i think id prefer to keep them as strings so the caller can reason about it easier